l1-contracts/contracts/bridge/L1AssetRouter.sol

```diff
@@ -607,7 +607,9 @@ contract L1AssetRouter is
         address l1AssetHandler = assetHandlerAddress[assetId];
         // slither-disable-next-line unused-return
         IL1AssetHandler(l1AssetHandler).bridgeMint(_chainId, assetId, transferData);
-        (amount, l1Receiver) = abi.decode(transferData, (uint256, address));
+        if (l1AssetHandler == address(nativeTokenVault)) {
+            (amount, l1Receiver) = abi.decode(transferData, (uint256, address));
+        }
 
         emit WithdrawalFinalizedSharedBridge(_chainId, l1Receiver, assetId, amount);
```

Comment:
the event will have broken values for non-NTV withdrawals. Considering that `WithdrawalFinalizedSharedBridge` is a relatively new method, are we sure there are people who rely on it?

I think we should change this event to be universal for all tokens